Playground/v2.1

kaushalbhardwaj opened the pull request:
Improvements in the playground 2.0

Covered all open issues

Dominik1999 commented:
Very nice, great job. A couple of points as feedback:

- **PR size:** Can you make smaller PRs (one per issue) instead of pushing those big PRs? Smaller PRs are easier to review
- **Font size:** Can you make the font size of the Inputs the same as the Outputs and Program Info?
- **Instructions:** Nice work with the instructions
- **Proving / Webworker:** Whenever I choose a new example, proving clearly takes longer than 1 second but the toast shows me something in milliseconds. See video attached:

https://github.com/0xPolygonMiden/examples/assets/35031754/799ec2a8-2147-4644-97d5-68210544d028

- **Game of Life / Catalan / Dft / Shamir (and others):** something is broken in the Game of Life, Catalan, Dft, and Shamir example, I get `Error: unreachable`. In those cases I can click `Debug` but nothing happens when I click the arrow buttons.
- **Display stack in Debug Mode:** In some cases, e.g. provable compliance, when the stack is displayed in debug, the values overlap and one cannot read them anymore. I think we should display the stack in two rows vertically.

![Screenshot 2024-07-03 at 09 40 42](https://github.com/0xPolygonMiden/examples/assets/35031754/94143609-f851-4e8d-a6db-63dd4edad407)

- **Colors:** Can you maybe change the green to the blue / lila that you use in the buttons. I think the playground has too many colors now

![Screenshot 2024-07-03 at 09 41 52](https://github.com/0xPolygonMiden/examples/assets/35031754/82125251-4c95-4482-a76d-db04c113981b)

kaushalbhardwaj commented:
PR size: Can you make smaller PRs (one per issue) instead of pushing those big PRs? Smaller PRs are easier to review- Can't segregate the changes now will be difficult 
Font size: Can you make the font size of the Inputs the same as the Outputs and Program Info? - Fixed
Proving / Webworker: Whenever I choose a new example, proving clearly takes longer than 1 second but the toast shows me something in milliseconds. - Changed the logic and fixed it 
Colors: Can you maybe change the green to the blue / lila that you use in the buttons. I think the playground has too many colors now- Replaced input green with accent color for consistency

Dominik1999 commented:
Very nice, thanks!

- **Colors:** Can you also make the color in the output box accent for consistency?
- **Error: unreachable:** I get this error on `a50d58f7` a lot of times. Can you re-check?
https://github.com/0xPolygonMiden/examples/assets/35031754/63a069cf-26f4-4d74-a827-01322984b888

Dominik1999 commented:
thanks a lot!

I get some linting issues 

```
[eslint] 
src/components/CodingEnvironment/MidenCode.tsx
  Line 75:40:  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  Line 75:45:  'viewUpdate' is defined but never used    @typescript-eslint/no-unused-vars
  Line 75:57:  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

src/pages/CodingEnvironment.tsx
  Line 831:36:  'backProps' is defined but never used  @typescript-eslint/no-unused-vars

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.

WARNING in [eslint] 
src/components/CodingEnvironment/MidenCode.tsx
  Line 75:40:  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  Line 75:45:  'viewUpdate' is defined but never used    @typescript-eslint/no-unused-vars
  Line 75:57:  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  ```
  
  And still the `Error: unreachable` persists (at least on my machine).

Dominik1999 commented:
![Screenshot 2024-07-16 at 11 44 39](https://github.com/user-attachments/assets/71e52d29-f42d-4074-8f3d-e2da21003e32)

when I try to load the instructions table, many instructions are empty. Do I need to run the script somehow?

Dominik1999 commented on playground/src/components/Header.tsx:
```diff
@@ -4,22 +4,22 @@ import { Dialog } from '@headlessui/react';
   {
-    name: 'Developer Tools',
-    href: 'https://0xpolygonmiden.github.io/miden-vm/tools/main.html'
+    name: 'DOCUMENTATION',
```
Can you change the documentation link to 

https://docs.polygon.technology/miden/

Dominik1999 reviewed (commented):
Looks good to me, left minor questions

kaushalbhardwaj commented on playground/src/data/index.js:
you need to run: npm install && node index.js

kaushalbhardwaj commented on playground/src/pages/CodingEnvironment.tsx:
```diff
@@ -359,13 +482,15 @@ export default function CodingEnvironment(): JSX.Element {
         try {
           const start = Date.now();
 
+          console.log('the data for program issss', code);
```
removed

kaushalbhardwaj commented on playground/src/pages/CodingEnvironment.tsx:
```diff
@@ -396,76 +526,71 @@ export default function CodingEnvironment(): JSX.Element {
-        if (!inputCheck.isValid) {
-          setOutput(inputCheck.errorMessage);
+    worker.onmessage = (e) => {
+      console.log('Worker message:', e.data); // Log the message for debugging
```
removed
